projects/api/__tests__/specs/auth/onboarding.ts

```diff
@@ -0,0 +1,111 @@
+import request, { Response } from 'supertest'
+import server from '../../../src/app'
+import { johnDoe } from '../../mocks/user'
+import { IUser } from '../../../src/interfaces'
+
+describe('POST /auth/onboarding', () => {
+    it('Wrong birthdate (year < 1900)', async () => {
+        const data: Partial<IUser> = {
+            avatar: 'placeholder',
+            username: 'TheTester',
+            birthdate: '01/01/1899',
+            description: 'My super description',
+        }
+
+        // Sending the data
+        const signupRes: Response = await request(server)
+            .post('/auth/signup')
+            .send(johnDoe)
+            .expect(200)
+        const token: string = signupRes.body.token
+
+        const onboardingRes: Response = await request(server)
+            .post('/auth/onboarding')
+            .set('Authorization', token)
+            .send(data)
+            .expect(400)
+
+        expect(onboardingRes.body.message).toBe('invalid_birthdate')
+    })
+
+    it('Wrong birthdate (year > actual year)', async () => {
+        const data: Partial<IUser> = {
+            avatar: 'placeholder',
+            username: 'TheTester',
+            birthdate: '01/01/2900',
+            description: 'My super description',
+        }
+
+        // Sending the data
+        const signupRes: Response = await request(server)
+            .post('/auth/signup')
+            .send(johnDoe)
+            .expect(200)
+        const token: string = signupRes.body.token
+
+        const onboardingRes: Response = await request(server)
+            .post('/auth/onboarding')
+            .set('Authorization', token)
+            .send(data)
+            .expect(400)
+
+        expect(onboardingRes.body.message).toBe('invalid_birthdate')
+
+    })
+
+    it('Wrong birthdate (incomplete)', async () => {
+        const data: Partial<IUser> = {
+            avatar: 'placeholder',
+            username: 'TheTester',
+            birthdate: '01/01/199_',
+            description: 'My super description',
+        }
+
+        // Sending the data
+        const signupRes: Response = await request(server)
+            .post('/auth/signup')
+            .send(johnDoe)
+            .expect(200)
+        const token: string = signupRes.body.token
+
+        const onboardingRes: Response = await request(server)
+            .post('/auth/onboarding')
+            .set('Authorization', token)
+            .send(data)
+            .expect(400)
+
+        expect(onboardingRes.body.message).toBe('invalid_birthdate')
+    })
+
+    it('Username already taken', async () => {
```

Comment:
I would put something like `Valid data successfully onboard and generates 'username_already_taken' on the second attempt`